compiler/cgen.nim

```diff
@@ -406,7 +406,12 @@ proc resetLoc(p: BProc, loc: var TLoc) =
   if isImportedCppType(typ): return
   if optSeqDestructors in p.config.globalOptions and typ.kind in {tyString, tySequence}:
     assert rdLoc(loc) != nil
-    linefmt(p, cpsStmts, "$1.len = 0; $1.p = NIM_NIL;$n", [rdLoc(loc)])
+
+    let atyp = skipTypes(loc.t, abstractInst)
+    if atyp.kind in {tyVar}:
```

Comment:
is `tyLent` also needed (I don't know, but the test should also check for lent to make sure it works)

Reply:
Ok, will check it after seeing whether CI is green.

Reply:
It doesn't seem to be needed.

This compiles before.
```nim
iterator mvalues(t: var seq[seq[int]]): lent seq[int] =
  yield t[0]

var t: seq[seq[int]]

while false:
  for v in t.mvalues:
    discard

proc ok =
  while false:
    for v in t.mvalues:
      discard

ok()

while false:
  for v in t.mvalues:
    discard
```

Reply:
It's safer to include the `tyLent` though.

Reply:
will do after CI is green.